### ISSUE TITLE
Minimum cmake version on OSX is wrong

### ISSUE THREAD

**Issue by kumpera** [VowpalWabbit/vowpal_wabbit#1684]
I was unable to build vw on OSX using cmake 3.5.1, but it works fine with 3.13.1

**Comment by jackgerrits** [comment]
The version of Boost that CMake is able to find is tied to the version of CMake that you are using. If the installation of Boost is sufficiently recent then CMake will fail to find it.

**Comment by JohnLangford** [comment]
Why does this happen?

**Comment by jackgerrits** [comment]
The CMake FindBoost module is updated as part of CMake. When Boost releases new versions it potentially needs to know how to handle new components. I don't know 100% thought, that's the best rationale I can come up with.

**Comment by kumpera** [comment]
It happens, as Jack explained to me earlier today, because cmake & boost have a mutual dependency among them. Meaning that if you have a recent boost version with an old cmake, the later won't be able to find the newer boost. 

And this plays straight into homebrew's lack of any versioning constraint on its packages dependencies. 
I got into this situation because I had an old version of cmake installed but had just updated boost.

To reproduce it, the follow should do it: 
```
brew install cmake boost
brew switch cmake 3.5.1 
```

I'll update README.md to have some troubleshooting help about this.

**Comment by JohnLangford** [comment]
Thank you.

Is there anything else we can do to mitigate this?

-John

**Comment by jackgerrits** [comment]
I agree, I'm going to reopen this and update the title to match this

**Comment by jackgerrits** [comment]
CMake prints `New Boost version may have incorrect or missing dependencies and imported targets` when it finds a version of Boost that is newer than it knows how to understand.

I think there is little we can do here, apart from making sure the documentation is clear and perhaps upgrading the minimum CMake version required.
